app/services/allocation_summary_service.rb

```diff
@@ -0,0 +1,132 @@
+class AllocationSummaryService
+  PAGE_SIZE = 10 # The number of items to show in the view
+  FETCH_SIZE = 500 # How many records to fetch from nomis at a time
+
+  # rubocop:disable Metrics/MethodLength
+  def summary(allocated_page, unallocated_page, missing_info_page, prison)
+    create_buckets(allocated_page, unallocated_page, missing_info_page)
+
+    # We want to store the total number of each item so we can show totals for
```

Comment:
The use of comments in the class is really helpful :-)
Only to points on this:
- Would be good to pass the initializer params as a hash in order to prevent any possible bugs from modifying the caller
- The test coverage is still 100% but I worried that if we modify this class further down the line, we do not have many tests to tell us if it is working as it should. I definitely don't want to add tests for the sake of tests but given this class is relatively complicated I wonder if testing unhappy paths could be a good idea